### ISSUE TITLE
Problem with edge creation using binary protocol in distributed mode

### ISSUE THREAD

**Issue by tobiemh** [orientechnologies/orientdb#3745]
```sql
CREATE CLASS Person extends V
CREATE CLASS Friend extends E

CREATE VERTEX Person SET fullname = 'Joe Bloggs'
CREATE VERTEX Person SET fullname = 'Tom Jones'
CREATE VERTEX Person SET fullname = 'Mark Zuckerberg'

CREATE EDGE Friend FROM ( SELECT FROM Person WHERE fullname = 'Joe Bloggs' ) TO ( SELECT FROM Person WHERE fullname = 'Tom Jones' )
```

The edge is created, and successfully copied to all distributed nodes, but the following error is raised on the server handling the query. This results in the binary protocol being out of sync, and all further queries fail.

```shell
orient|2015-03-10 22:22:53:928 INFO  [core-5] Saving distributed configuration file for database 'testdatabase' to: /opt/orient/databases/testdatabase/distributed-config.json [OHazelcastPlugin]Error on unmarshalling record #33:0 (com.orientechnologies.orient.core.exception.ODatabaseException: Impossible save a record during network serialization)
orient|com.orientechnologies.orient.core.exception.ODatabaseException: Impossible save a record during network serialization
orient| at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerCSVAbstract.linkToStream(ORecordSerializerCSVAbstract.java:132)
orient| at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerCSVAbstract.fieldToStream(ORecordSerializerCSVAbstract.java:364)
orient| at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerSchemaAware2CSV.toString(ORecordSerializerSchemaAware2CSV.java:506)
orient| at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerStringAbstract.toStream(ORecordSerializerStringAbstract.java:688)
orient| at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerSchemaAware2CSV.toStream(ORecordSerializerSchemaAware2CSV.java:268)
orient| at com.orientechnologies.orient.core.record.impl.ODocument.toStream(ODocument.java:2057)
orient| at com.orientechnologies.orient.core.record.impl.ODocument.toStream(ODocument.java:682)
orient| at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.getRecordBytes(OBinaryNetworkProtocolAbstract.java:411)
orient| at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.writeRecord(OBinaryNetworkProtocolAbstract.java:426)
orient| at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.writeIdentifiable(OBinaryNetworkProtocolAbstract.java:139)
orient| at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1216)
orient| at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:385)
orient| at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
orient| at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
```

Subsequent requests produce errors similar to this...

```shell
orient|2015-03-10 22:26:46:048 WARNING Cannot serialize record: e[#32:0][#28:0-Friend->#28:1] [ONetworkProtocolBinary]Error on unmarshalling record #32:1 (com.orientechnologies.orient.core.exception.ODatabaseException: Impossible save a record during network serialization)
orient|com.orientechnologies.orient.core.exception.ODatabaseException: Impossible save a record during network serialization
orient|  at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerCSVAbstract.linkToStream(ORecordSerializerCSVAbstract.java:132)
orient|  at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerCSVAbstract.fieldToStream(ORecordSerializerCSVAbstract.java:364)
orient|  at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerSchemaAware2CSV.toString(ORecordSerializerSchemaAware2CSV.java:506)
orient|  at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerStringAbstract.toStream(ORecordSerializerStringAbstract.java:688)
orient|  at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerSchemaAware2CSV.toStream(ORecordSerializerSchemaAware2CSV.java:268)
orient|  at com.orientechnologies.orient.core.record.impl.ODocument.toStream(ODocument.java:2057)
orient|  at com.orientechnologies.orient.core.record.impl.ODocument.toStream(ODocument.java:682)
orient|  at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.getRecordBytes(OBinaryNetworkProtocolAbstract.java:411)
orient|  at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.writeRecord(OBinaryNetworkProtocolAbstract.java:426)
orient|  at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.writeIdentifiable(OBinaryNetworkProtocolAbstract.java:139)
orient|  at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1216)
orient|  at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:385)
orient|  at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:220)
orient|  at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
```

I have tried using the following distributed configurations:

```json
{
    "autoDeploy": true,
    "hotAlignment": true,
    "readQuorum": 1,
    "writeQuorum": 1,
    "failureAvailableNodesLessQuorum": false,
    "readYourWrites": true,
    "clusters": {
        "internal": {
        },
        "index": {
        },
        "*": {
            "servers" : [ "<NEW_NODE>" ]
        }
    }
}
```

AND

```json
{
    "autoDeploy": true,
    "hotAlignment": true,
    "readQuorum": 1,
    "writeQuorum": 2,
    "failureAvailableNodesLessQuorum": false,
    "readYourWrites": true,
    "clusters": {
        "internal": {
        },
        "index": {
        },
        "*": {
            "servers" : [ "<NEW_NODE>" ]
        }
    }
}
```

NOTE: The same queries work using REST and/or the Orient Studio without issues.

NOTE: The same queries work in standalone mode.

***

OrientDB Version: 2.0.3 AND 2.0.4

**Comment by lvca** [comment]
Fixed in 2.0.5-SNAPSHOT. Please let me know if works, so I can release 2.0.5 asap.

**Comment by tobiemh** [comment]
Amazing... I only just added the issue! Yes I can confirm that this works in 2.0.5-SNAPSHOT.

Thanks @lvca
